pkg/kube/kube_handler.go

```diff
@@ -726,6 +727,266 @@ func doValidationRequest(cli httpclient.Client, req *http.Request) (*ValidationR
+	// This fuction is the implementation of below curl command
+	// curl -XGET -H "Authorization: Basic $harborauthz"
+	//		 -H "Accept: application/vnd.oci.image.manifest.v1+json"
+	//		-s https://demo.goharbor.io/v2/test10/podinfo/podinfo/manifests/6.0.0
+
+	parsedURL, err := url.ParseRequestURI(repoURL)
+	if err != nil {
+		return nil, err
+	}
+	parsedURL.Path = path.Join("v2", parsedURL.Path, repoName, "manifests", tagVersion)
+
+	log.Infof("parsedURL %v", parsedURL.String())
+	req, err := http.NewRequest("GET", parsedURL.String(), nil)
+	if err != nil {
+		return nil, err
+	}
+	//TODO: remove Auth header
+	req.Header.Set("Authorization", os.ExpandEnv("Basic $harborauthz"))
+	req.Header.Set("Accept", "application/vnd.oci.image.manifest.v1+json")
+
+	resp, err := cli.Do(req)
+
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	var mediaData ConfigMediaType
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	err = json.Unmarshal(body, &mediaData)
+	if err != nil {
+		return nil, err
+	}
+	mediaType := mediaData.Config.MediaType
+	return &mediaType, nil
+}
+
+//compareMediaType matches wantStr with the inputStr
+func compareMediaType(repoName string, inputStr string, wantStr string) (bool, error) {
+
+	err := fmt.Errorf("%v is not a Helm OCI Repo", repoName)
+	result := false
+
+	tmpStr1 := regexp.MustCompile(`[\\/\\+]`).Split(inputStr, -1)
+	if len(tmpStr1) < 2 {
+		return result, err
+	}
+
+	tmpStr2 := regexp.MustCompile(`.[^.]*$`).Split(tmpStr1[1], -1)
+	if len(tmpStr2) == 0 {
+		return result, err
+	}
+	gotStr := tmpStr2[0]
+
+	log.Infof("gotStr : (%v)\n", gotStr)
+	if gotStr != wantStr {
+		return result, err
+	}
+	return true, nil
+}
+
+// ValidateOCIAppRepository  validates OCI Repos only
+func ValidateOCIAppRepository(appRepo *v1alpha1.AppRepository, cli httpclient.Client) (bool, error) {
+
+	var err error
+
+	result := false
+	err = nil
+
+	repoURL := strings.TrimSuffix(strings.TrimSpace(appRepo.Spec.URL), "/")
+
+	// For the OCI case, we want to validate that all the given repositories are valid
+	if len(appRepo.Spec.OCIRepositories) == 0 {
+		return false, ErrEmptyOCIRegistry
+	}
+	for _, repoName := range appRepo.Spec.OCIRepositories {
+		log.Infof("repoURL : %v, repoName : %v\n", repoURL, repoName)
+		tagVersionPtr, err := getOCIAppRepositoryTag(cli, repoURL, repoName)
+		if err != nil {
+			return false, err
+		}
+
+		log.Infof("tagVersion: %v", *tagVersionPtr)
+
+		mediaTypePtr, err := getOCIAppRepositoryMediaType(cli, repoURL, repoName, *tagVersionPtr)
+		if err != nil {
+			return false, err
+		}
+
+		log.Infof("mediaType : %v", *mediaTypePtr)
+
+		wantStr := "vnd.cncf.helm.config"
+
+		result, err = compareMediaType(repoName, *mediaTypePtr, wantStr)
+		if !result {
+			return result, err
+		}
+	}
+	return result, err
+}
+
+type nonOCIRepoRequests struct {
+	reqs []*http.Request
+}
+
+type OCIRepoRequests struct {
+	appRepo *v1alpha1.AppRepository
+}
+
```

Comment:
```suggestion
// HttpValidator validates a Helm repository using HTTP requests
```

Also, not a big deal but normally you'd define the interface first, then each implementation (struct, function). Fine as is though.